File: Titanium/Décision/Monte_Carlo/templates/carlo.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func monteCarloSimulation(nSimulations int) float64 {
	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
	pointsInsideCircle := 0

	for i := 0; i < nSimulations; i++ {
		x, y := rng.Float64(), rng.Float64()
		if x*x+y*y <= 1 {
			pointsInsideCircle++
		}
	}

	return 4 * float64(pointsInsideCircle) / float64(nSimulations)
}

func main() {
	nSimulations := 100_000 // Go 1.13+ permet le séparateur '_'
	piEstimate := monteCarloSimulation(nSimulations)
	fmt.Printf("Estimation de π : %.6f\n", piEstimate)  // Estimation de π : 3.133920
} 

File: Titanium/Décision/Monte_Carlo/templates/carlo.py
import numpy as np

def monte_carlo_simulation(n_simulations: int) -> float:
    """
    Estime la valeur de π en utilisant la méthode de Monte-Carlo.
    """
    # Générer des points aléatoires en une seule opération pour optimiser la performance
    points = np.random.rand(n_simulations, 2)
    
    # Calculer le nombre de points à l'intérieur du cercle
    points_inside_circle = np.sum(np.sum(points**2, axis=1) <= 1)
    
    # Estimation de π
    return 4 * points_inside_circle / n_simulations

if __name__ == "__main__":
    n_simulations = 100_000  # Amélioration de la lisibilité avec un séparateur
    pi_estimate = monte_carlo_simulation(n_simulations)
    print(f"Estimation de π : {pi_estimate:.6f}")  # Estimation de π : 3.142880 
